src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -614,6 +614,23 @@ public ProgrammingExercise findById(Long programmingExerciseId) throws EntityNot
         }
     }
 
+    /**
+     * Find a programming exercise by its id, with eagerly loaded studentParticipations.
+     *
+     * @param programmingExerciseId of the programming exercise.
+     * @return The programming exercise related to the given id
+     * @throws EntityNotFoundException the programming exercise could not be found.
+     */
+    public ProgrammingExercise findByIdWithEagerStudentParticipations(Long programmingExerciseId) throws EntityNotFoundException {
```

Comment:
```suggestion
    public ProgrammingExercise findByIdWithEagerStudentParticipations(long programmingExerciseId) throws EntityNotFoundException {
```

Might be better to avoid passing `null` in the first place

Reply:
Good point!